packages/lib-stellar/index.ts

```diff
@@ -0,0 +1,81 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+type WalletMetadata = GetWalletResponse
+
+const base64ToHex = (base64: string): string => {
+  return Buffer.from(base64, 'base64').toString('hex')
+}
+
+const hexToBase64 = (hex: string): string => {
+  return Buffer.from(stripHexPrefix(hex), 'hex').toString('base64')
+}
+
+const stripHexPrefix = (hex: string): string => {
+  return hex.replace(/^0x/, '')
+}
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Stellar' && res.network !== 'StellarTestnet') {
+      throw new DfnsError(-1, 'wallet is not bound to Stellar or StellarTestnet', {
+        walletId,
+        network: res.network,
+      })
+    }
+
+    return new DfnsWallet(res, options)
+  }
+
+  public get address(): string {
+    return this.metadata.address!
+
+  }
+
+  public async sign(unsignedTxXdr: string): Promise<string> {
```

Comment:
let's make this a little friendlier, we can sign the actual transaction objects and overload this for better typing.

```
public async sign(transaction: Transaction): Promise<Transaction>
public async sign(transaction: FeeBumpTransaction): Promise<Transaction>
public async sign(transaction: Transaction | FeeBumpTransaction): Promise<Transaction | FeeBumpTransaction> {
  // code here
}
```

Reply:
I understand what you mean here, I chose first to use unsignedTxXdr because it was friendly with the stellar laboratory for example. But I will do that this way ^^^^